README.md
# iOS - Intro Screens

This demo provides Introduction screens in scrollview manner usually used at the startup of application. It manages dynamic views based on input images.

## Table of contents
- [iOS Support](#iOS-support)
- [Demo](#demo)
- [Features](#features)
- [Getting started](#getting-started)
- [Usage](#usage)
- [Methods](#methods)
- [Want to Contribute?](#want-to-contribute)
- [Need Help / Support?](#need-help)
- [Collection of Components](#collection-of-Components)
- [Changelog](#changelog)
- [License](#license)
- [Keywords](#Keywords)


## iOS Support

Versions: iOS 12, iOS 13

We have tested our program in above versions, however you can use it in other versions as well.

## Demo
[![](IntroductionScreens.gif)](https://github.com/weblineindia/iOS-Intro-Screens/blob/master/IntroductionScreens.gif)

------

## Features

* Introduction Screen demo app uses Scrollview and UIPageControl with Skip feature.
* This is a reusable component and you can add it in your iOS application.

## Getting started

Download this project and import 'App_Startup_gallery' folder contents in your project.

## Usage

### import App_Startup_gallery folder in your project 

* Make sure 'Images' folder should be added as 'group by reference' in your sample project.
* Setup below configuration in your root viewcontroller 

### Navigate to Startup component from your root viewcontroller.

e.g.

        let storyboard = UIStoryboard(name: "AppStartUp", bundle: nil)
        let secondVC = storyboard.instantiateViewController(identifier: "StartupGallery")
        
### This component use Images from "Images" folder", We can replace or add new images into it 


## Methods

### Below method for getting images from folder
e.g.

        let fm = FileManager.default
        let path = Bundle.main.resourcePath!
        print(path)
        do{
        let item = try fm.contentsOfDirectory(atPath: path).filter{$0.lowercased().hasSuffix(".png") || $0.lowercased().hasSuffix(".jpg")}
        arrImages.removeAll()
        for imgs in item{
            print(imgs)
            arrImages.append(UIImage(named:imgs)!)
            print(arrImages)
        }
        }catch let e{
            print("error  : \(e)")
        }
        
### Below method for loading images in Scrollview.
e.g.

     func loadScrollView() 

### Below Delegate method for getting current page number. 
e.g.

    func scrollViewDidEndDecelerating(_ scrollView: UIScrollView) {
    }
    
### Use this method for adding target on page control 
e.g.

    self.pageControl.addTarget(self, action: #selector(self.pageChanged(sender:)), for: UIControl.Event.valueChanged)
    @objc func pageChanged(sender:AnyObject)
       {
           let xVal = CGFloat(pageControl.currentPage) * imgScrollView.frame.size.width
           imgScrollView.setContentOffset(CGPoint(x: xVal, y: 0), animated: true)
           
       }
       
 ------
 
## Want to Contribute?

- Created something awesome, made this code better, added some functionality, or whatever (this is the hardest part).
- [Fork it](http://help.github.com/forking/).
- Create new branch to contribute your changes.
- Commit all your changes to your branch.
- Submit a [pull request](http://help.github.com/pull-requests/).

 ------

## Need Help? 

We also provide a free, basic support for all users who want to use this Intro Screens demo in project. In case you want to customize this code library to suit your development needs, then feel free to contact our [iOS
developers](https://www.weblineindia.com/hire-ios-app-developers.html).

 ------
 
## Collection of Components

We have built many other components and free resources for software development in various programming languages. Kindly click here to view our [Free Resources for Software Development.](https://www.weblineindia.com/software-development-resources.html)

------

## Changelog

 Detailed changes for each release are documented in [CHANGELOG](./CHANGELOG).

## License

 [MIT](LICENSE)

 [mit]: https://github.com/weblineindia/iOS-Intro-Screens/blob/master/LICENSE

## Keywords

App Introduction pages, Startup pages, Introduction screens, ScrollView images, PageControl gallery.


CHANGELOG
1.0.1 (July 10, 2020)
Initial release - Sample source code for Introduction StartUp pages.


App_Startup_gallery/AppDelegate.swift
//
//  AppDelegate.swift
//  App_Startup_gallery
//
//  Created by WeblineIndia  on 08/07/20.
//  Copyright © 2020 WeblineIndia . All rights reserved.
//

import UIKit

@available(iOS 13.0, *)
@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {



    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        return true
    }

    // MARK: UISceneSession Lifecycle

    func application(_ application: UIApplication, configurationForConnecting connectingSceneSession: UISceneSession, options: UIScene.ConnectionOptions) -> UISceneConfiguration {
        // Called when a new scene session is being created.
        // Use this method to select a configuration to create the new scene with.
        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
    }

    func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
        // Called when the user discards a scene session.
        // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
        // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
    }


}



App_Startup_gallery/SceneDelegate.swift
//
//  SceneDelegate.swift
//  App_Startup_gallery
//
//  Created by WeblineIndia  on 08/07/20.
//  Copyright © 2020 WeblineIndia . All rights reserved.
//

import UIKit

@available(iOS 13.0, *)
class SceneDelegate: UIResponder, UIWindowSceneDelegate {

    var window: UIWindow?


    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
        // Use this method to optionally configure and attach the UIWindow `window` to the provided UIWindowScene `scene`.
        // If using a storyboard, the `window` property will automatically be initialized and attached to the scene.
        // This delegate does not imply the connecting scene or session are new (see `application:configurationForConnectingSceneSession` instead).
        guard let _ = (scene as? UIWindowScene) else { return }
    }

    func sceneDidDisconnect(_ scene: UIScene) {
        // Called as the scene is being released by the system.
        // This occurs shortly after the scene enters the background, or when its session is discarded.
        // Release any resources associated with this scene that can be re-created the next time the scene connects.
        // The scene may re-connect later, as its session was not neccessarily discarded (see `application:didDiscardSceneSessions` instead).
    }

    func sceneDidBecomeActive(_ scene: UIScene) {
        // Called when the scene has moved from an inactive state to an active state.
        // Use this method to restart any tasks that were paused (or not yet started) when the scene was inactive.
    }

    func sceneWillResignActive(_ scene: UIScene) {
        // Called when the scene will move from an active state to an inactive state.
        // This may occur due to temporary interruptions (ex. an incoming phone call).
    }

    func sceneWillEnterForeground(_ scene: UIScene) {
        // Called as the scene transitions from the background to the foreground.
        // Use this method to undo the changes made on entering the background.
    }

    func sceneDidEnterBackground(_ scene: UIScene) {
        // Called as the scene transitions from the foreground to the background.
        // Use this method to save data, release shared resources, and store enough scene-specific state information
        // to restore the scene back to its current state.
    }


}



App_Startup_gallery/NextViewController.swift
//
//  NextViewController.swift
//  App_Startup_gallery
//
//  Created by WeblineIndia  on 08/07/20.
//  Copyright © 2020 WeblineIndia . All rights reserved.
//

import UIKit

class NextViewController: UIViewController {

    @IBOutlet weak var lblWelcome: UILabel!
    override func viewDidLoad() {
        super.viewDidLoad()
        self.navigationController?.navigationBar.isHidden = false
        lblWelcome.text = "Welcome!!"
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}


App_Startup_gallery/StartUpView.swift
//
//  ViewController.swift
//  App_Startup_gallery
//
//  Created by WeblineIndia  on 08/07/20.
//  Copyright © 2020 WeblineIndia. All rights reserved.
//

import UIKit

class StartUpView: UIViewController, UIScrollViewDelegate {
    
    @IBOutlet weak var baseView: UIView!
    @IBOutlet weak var imgScrollView: UIScrollView!
    @IBOutlet weak var pageControl: UIPageControl!
    @IBOutlet weak var btnPrevious: UIButton!
    @IBOutlet weak var btnSkip: UIButton!
    @IBOutlet weak var btnNext: UIButton!
    @IBOutlet weak var btnContinue: UIButton!
    var arrImages : [UIImage] = []
    override func viewDidLoad() {
        super.viewDidLoad()
       
        // get images from path and add it in array
        let fm = FileManager.default
        let path = Bundle.main.resourcePath! + "/Images"
           
        print(path)
        do{
            let item = try fm.contentsOfDirectory(atPath: path).filter{$0.lowercased().hasSuffix(".png") || $0.lowercased().hasSuffix(".jpg")}
            print(item)
            arrImages.removeAll()
            for imgs in item{
                print(imgs)
                arrImages.append(UIImage(named:"Images/\(imgs)")!)
                print(arrImages)
            }
        }catch let e{
            print("error  : \(e)")
        }
        btnNext.isHidden = true
        btnPrevious.isHidden = true
        btnSkip.roundCorners()
        btnContinue.roundCorners()
        loadScrollView()
    }
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        self.navigationController?.navigationBar.isHidden = true
    }
    // Loading image in scrollview
    func loadScrollView() {
        let pageCount = arrImages.count
        imgScrollView.frame = view.bounds
        imgScrollView.delegate = self
        imgScrollView.backgroundColor = UIColor.clear
        imgScrollView.isPagingEnabled = true
        imgScrollView.showsHorizontalScrollIndicator = false
        imgScrollView.showsVerticalScrollIndicator = false
        pageControl.numberOfPages = pageCount
        pageControl.currentPage = 0
        
        for i in (0..<pageCount) {
            let imageView = UIImageView()
            imageView.frame = CGRect(x: i * Int(self.view.frame.size.width) , y: 0 , width:
                Int(self.view.frame.size.width) , height: Int(self.view.frame.size.height/1.2))
            imageView.contentMode = .scaleAspectFit
            imageView.image = arrImages[i]
//            imageView.backgroundColor = UIColor.white
            self.imgScrollView.addSubview(imageView)
        }
        
        let width1 = (Float(arrImages.count) * Float(self.view.frame.size.width))
        imgScrollView.contentSize = CGSize(width: CGFloat(width1), height: self.view.frame.size.height)
        self.pageControl.addTarget(self, action: #selector(self.pageChanged(sender:)), for: UIControl.Event.valueChanged)
                
    }
    
    //MARK: -
    // Scrollview Delegate method
    func scrollViewDidScroll(_ scrollView: UIScrollView) {
        imgScrollView.contentOffset.y = 0
    }
    func scrollViewDidEndDecelerating(_ scrollView: UIScrollView) {
        let pageNumber = round(scrollView.contentOffset.x / scrollView.frame.size.width)
        pageControl.currentPage = Int(pageNumber)
        print(pageNumber)
        if  pageControl.currentPage == arrImages.count - 1{
            btnContinue.setTitle("Finish", for: .normal)
        }
        else{
            btnContinue.setTitle("Continue", for: .normal)
        }
        
        
    }
    
    @objc func pageChanged(sender:AnyObject)
    {
        let xVal = CGFloat(pageControl.currentPage) * imgScrollView.frame.size.width
        imgScrollView.setContentOffset(CGPoint(x: xVal, y: 0), animated: true)
        
    }
    
    //change page number on click of button
    func changedPageNumber(){
        let pageNumber = round(imgScrollView.contentOffset.x / imgScrollView.frame.size.width)
        pageControl.currentPage = Int(pageNumber)
        print(pageNumber)
        if  pageControl.currentPage == arrImages.count - 1{
            btnContinue.setTitle("Finish", for: .normal)
        }
        else{
            btnContinue.setTitle("Continue", for: .normal)
        }
    }
    
    // On tap display previous image
    @IBAction func btnPreviousTapped(_ sender: Any){
        
        if pageControl.currentPage > 0 {
            UIView.animate(withDuration: 0.5, delay: 0, options: .curveLinear, animations: {
                self.imgScrollView.contentOffset.x = self.imgScrollView.contentOffset.x - self.view.frame.size.width
            })
            
           changedPageNumber()
            
        }
    }
    
     // On tap display next image
    @IBAction func btnNextTapped(_ sender: Any) {
        
        if pageControl.currentPage < self.arrImages.count - 1 {
            UIView.animate(withDuration: 0.5, delay: 0, options: .curveLinear, animations: {
                self.imgScrollView.contentOffset.x = self.imgScrollView.contentOffset.x + self.view.frame.size.width
            })
            
            changedPageNumber()
        }
    }
    //On skip load welcome page
    @IBAction func btnSkipTapped(_ sender: Any) {
        let loginVC = UIStoryboard(name: "AppStartUp", bundle: nil).instantiateViewController(withIdentifier: "WelcomePage") as? NextViewController
        self.navigationController?.pushViewController(loginVC!, animated: true)
        
    }
    // On continue it will show next image
    @IBAction func btnContinueTapped(_ sender: Any) {
        if btnContinue.titleLabel?.text == "Finish"{
            let loginVC = UIStoryboard(name: "AppStartUp", bundle: nil).instantiateViewController(withIdentifier: "WelcomePage") as? NextViewController
            self.navigationController?.pushViewController(loginVC!, animated: true)
            
        }
        else{
            if pageControl.currentPage < self.arrImages.count - 1 {
                UIView.animate(withDuration: 0.5, delay: 0, options: .curveLinear, animations: {
                self.imgScrollView.contentOffset.x = self.imgScrollView.contentOffset.x + self.view.frame.size.width
                })
               changedPageNumber()
            }
        }
    }
}
extension UIButton {
    func roundCorners() {
        self.layer.cornerRadius = self.frame.size.height / 2
        self.clipsToBounds = true
    }
    
}
